VSDiagnostics/VSDiagnostics/VSDiagnostics.Test/Tests/Attributes/EnumCanHaveFlagsAttributeTests.cs

```diff
@@ -0,0 +1,139 @@
+﻿using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using RoslynTester.Helpers.CSharp;
+using VSDiagnostics.Diagnostics.Attributes.EnumCanHaveFlagsAttribute;
+
+namespace VSDiagnostics.Test.Tests.Attributes
+{
+    [TestClass]
+    public class EnumCanHaveFlagsAttributeTests : CSharpCodeFixVerifier
+    {
+        protected override DiagnosticAnalyzer DiagnosticAnalyzer => new EnumCanHaveFlagsAttributeAnalyzer();
+
+        protected override CodeFixProvider CodeFixProvider => new EnumCanHaveFlagsAttributeCodeFix();
+
+        [TestMethod]
+        public void EnumCanHaveFlagsAttribute_AddsFlagsAttribute()
+        {
+            var original = 
+@"namespace ConsoleApplication1
+{
+    enum Foo
+    {
+    }
+}";
+
+            var result = 
+@"using System;
+
+namespace ConsoleApplication1
+{
+    [Flags]
+    enum Foo
+    {
+    }
+}";
+
+            VerifyDiagnostic(original, EnumCanHaveFlagsAttributeAnalyzer.Rule.MessageFormat.ToString());
+            VerifyFix(original, result);
+        }
+
+        [TestMethod]
+        public void EnumCanHaveFlagsAttribute_AddsFlagsAttribute_DoesNotAddDuplicateUsingSystem()
+        {
+            var original =
+@"using System;
+
+namespace ConsoleApplication1
+{
+    enum Foo
+    {
+    }
+}";
+
+            var result =
+@"using System;
+
+namespace ConsoleApplication1
+{
+    [Flags]
+    enum Foo
+    {
+    }
+}";
+
+            VerifyDiagnostic(original, EnumCanHaveFlagsAttributeAnalyzer.Rule.MessageFormat.ToString());
+            VerifyFix(original, result);
+        }
+
+        [TestMethod]
+        public void EnumCanHaveFlagsAttribute_AddsFlagsAttribute_OnlyAddsFlagsAttribute()
+        {
+            var original =
+@"using System;
+
+namespace ConsoleApplication1
+{
+    [Obsolete(""I'm obsolete"")]
+    enum Foo
+    {
+        Goo = 0,
+        Hoo,
+        Joo,
+        Koo,
+        Loo
+    }
+}";
+
+            var result =
+@"using System;
+
+namespace ConsoleApplication1
+{
+    [Obsolete(""I'm obsolete"")]
+    [Flags]
+    enum Foo
+    {
+        Goo = 0,
+        Hoo,
+        Joo,
+        Koo,
+        Loo
+    }
+}";
+
+            VerifyDiagnostic(original, EnumCanHaveFlagsAttributeAnalyzer.Rule.MessageFormat.ToString());
+            VerifyFix(original, result);
+        }
+
+        [TestMethod]
+        public void EnumCanHaveFlagsAttribute_AddsFlagsAttribute_AddsUsingSystemWhenUsingSystemDotAnything()
+        {
+            var original =
+@"using System.Text;
+
+namespace ConsoleApplication1
+{
+    enum Foo
+    {
+    }
+}";
+
+            var result =
+@"using System;
+using System.Text;
+
+namespace ConsoleApplication1
+{
+    [Flags]
+    enum Foo
+    {
+    }
+}";
+
+            VerifyDiagnostic(original, EnumCanHaveFlagsAttributeAnalyzer.Rule.MessageFormat.ToString());
+            VerifyFix(original, result);
+        }
+    }
+}
```

Comment:
Can you add a test that verifies it doesn't add the attribute when it's part of a chained attribute like this: `[Obsolete("reason"), Flags]`. I'm pretty sure it won't give a problem with the current implementation, but I'd like to have this scenario covered as well. Likewise a test that uses `[FlagsAttribute]` would be useful as well.